### PR TITLE
GDP: rename chull transformation to hull

### PR DESCRIPTION
**Pull request by qtothec**
## Summary/Motivation:
The hull reformulation (now at `'gdp.hull'`) has long been improperly named `gdp.chull`, but the HR may not result in a convex hull absent additional assumptions on the original GDP model. Empirically, it seems possible to obtain an exact reformulation for a nonconvex GDP using HR under certain conditions, resulting in a nonconvex MINLP. However, the theoretical implications remain to be established. @bernalde is best-suited to comment on this.

### Legal Acknowledgement

By contributing to this software project, I have read the [contribution guide](https://pyomo.readthedocs.io/en/stable/contribution_guide.html) and agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the BSD license.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: pyomo/gdp/plugins/hull.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ class ConvexHull_Transformation(Transformation):
         list of blocks and Disjunctions [default: the instance]
 
     The transformation will create a new Block with a unique
-    name beginning "_pyomo_gdp_chull_relaxation".  That Block will
+    name beginning "_pyomo_gdp_hull_relaxation".  That Block will
```
Comment (jsiirola) [inline pyomo/gdp/plugins/hull.py:67]: While we are renaming things, this should be renamed `_pyomo_gdp_hull_reformulation`

**Comment by qtothec** [comment]
I think I'm going to need to do a lot more renaming than originally intended now.

**Comment by qtothec** [comment]
Not sure what the Jenkins failure is, but this is now ready to review again.

**Comment by jsiirola** [comment]
The Jenkins test failures are unrelated to this PR and were resolved by #1473.  This should be safe to merge.

**Pull request merged**